### PR TITLE
Fetch networks from the server

### PR DESCRIPTION
**Pull request by yujiberra**
This PR adds front end functionality to get networks from the server by choosing an mRNA sample group, a miRNA sample group, and a gene set.

The fetched network isn't visualized in any way, but it's accessible in the `fetchedNetwork$` property of `NetworkDisplayComponent`, and I also display it in the template, though it just shows up as [Object object] because it's just a basic JS object without a bespoke `toString` method. (I wanted to JSON.stringify it but that would have required me to add more code we wouldn't ultimately need).

Also, this PR is a bit weird because it's not a PR against master, but against `visualization-working`, which I'd like to use as a working branch for the network visualization. `visualization-working` right now is just `rodolfo/addVizComponent` with master (~~as of a day ago~~ latest as of 4 Jan) merged into it. I set things up this way since
- I think it would be good to get network visualization basically working overall before we merge it into master
- I'd like this PR to be reviewable as a unit
- I don't want the `master->rodolfo/addVizComponent` diff to show up in this PR (since that would make it hard to review)

Some other changes I made:
- added a `type` property to gene sets, to indicate their omics type (`mRNA` or `miRNA`)
  - please note that this means you won't be able to generate networks using gene sets that were made before merging this PR
- some minor fixes (tests, linter errors) ead6847e9d82aee69ef9406bffa62aa94532591a

I tested this PR by fetching networks using both mRNA and miRNA gene sets, and making sure that the fetched network made sense (I used gene sets with only 1 or 2 probes, and made sure those probes showed up in the network's nodes, and that the interactions always included those probes).

### CODE REVIEW

**Comment by yujiberra** [comment]
@RodolfoAllendes 
Hey, so this PR should have what you need in order to start working on the network visualization.

My suggestion for how to move forward would be the following, but please let me know if you have other suggestions!
- If you could take a look and make comments/request changes I will make whatever updates you might need. 
- Then, I'll merge this PR 
- For your work on visualizing the network obtained from the back end, you can work off the `visualization-working` branch and then make a PR against it once you're ready

Also, while working on this I realized that it might be a good idea to set up some tooling to make code style consistent. I'm very much open to suggestions, but if possible I'd like us to try to follow the rules set up in https://github.com/Toxygates/Panomicon/blob/master/angular/.eslintrc.json (if you're using VS Code, there's an eslint plugin that will highlight errors in the editor)

For indentation as well, could we indent with spaces (and use 2 spaces per tab)? I realize that my existing HTML templates were indented with 4 spaces, but I'd like to switch them to 2 spaces. There's a nice tool called `prettier` that can be set up to automatically fix indentation and make other changes; I'm planning on making a PR to set that up, and I'll also simultaneously unify the indentation across the project when I do so.

**Comment by RodolfoAllendes** [comment]
Thank you for all your work on this @yujiberra 

I think not having an authentication service running (I guess getting the docker version working would be the easiest way to solve this), the 'Admin' tab is redirected to the page not found component, which in turn does not let me change the data already uploaded to the back-end... however, I was still able to generate some networks with the data that I already have here (I slightly remember you guys helping me with this some time ago). 

The network data itself looks good, and since it follows the structure we discussed before, at this point I do not have any further comments on that. I will now get on getting the display panel to actually show the network, rather than that [object Object] message ;)

As for the way to work with this, I do not have any problems with branching out the main `visualization-working` branch... will only wait for this code to actually be merged into that before doing so though ;)

As for coding styles... I have used eslint and prettier before, so feel free to add them as dev dependencies for the front-end... as I side note, I do like 2 spaces better than 4... although not particularly opiniated on the tab vs space use... as long as they are consistent through the code :)

**Review by RodolfoAllendes — APPROVED** [review]

**Comment by yujiberra** [comment]
@RodolfoAllendes 
Great, thanks for the review! 

Ah right, it makes sense that if you tested the branch for this PR you'd be able to use the viewer, but not the admin functionality, without FusionAuth. Since I've merged https://github.com/Toxygates/Panomicon/pull/5 into master, and I also merged master into `visualization-working`, for the next step you'll need to have FusionAuth running to even use the viewer, but once you get a version of Docker running that has the `docker-compose` command it really should just be a matter of doing `docker-compose up`.

Understood about eslint and prettier then, I'll start working on that soon. I too prefer 2 spaces to 4 as well (I'm not sure why the HTML templates in this project ended up with 4-space tabs - I was pretty new to VS Code when I was setting up this project, so maybe it was a default I forgot to change), so I guess we're in good shape :-D

I'll merge this now then; if you have any issues setting up FusionAuth, or with anything else let me know and I'll be happy to help!

**Pull request merged**
